daffodil-runtime2/src/test/resources/org/apache/daffodil/runtime2/ex_nums.tdml

```diff
@@ -37,62 +37,69 @@
   </tdml:defineConfig>
 
   <tdml:parserTestCase
-    name="ex_nums_parse_runtime1"
     model="ex_nums.dfdl.xsd"
+    name="ex_nums_runtime1"
     config="config-runtime1">
     <tdml:document>
-      <tdml:documentPart type="file">ex_nums_parse.dat</tdml:documentPart>
+      <tdml:documentPart type="file">ex_nums.dat</tdml:documentPart>
     </tdml:document>
     <tdml:infoset>
-      <tdml:dfdlInfoset type="file">ex_nums_unparse_runtime1.xml</tdml:dfdlInfoset>
+      <tdml:dfdlInfoset type="file">ex_nums.dat.xml.runtime1</tdml:dfdlInfoset>
     </tdml:infoset>
   </tdml:parserTestCase>
 
-  <tdml:unparserTestCase
-    name="ex_nums_unparse_runtime1"
-    model="ex_nums.dfdl.xsd"
-    config="config-runtime1">
-    <tdml:infoset>
-      <tdml:dfdlInfoset type="file">ex_nums_unparse_runtime1.xml</tdml:dfdlInfoset>
-    </tdml:infoset>
-    <tdml:document>
-      <tdml:documentPart type="file">ex_nums_parse.dat</tdml:documentPart>
-    </tdml:document>
-  </tdml:unparserTestCase>
-
   <tdml:parserTestCase
-    name="ex_nums_parse_runtime2"
     model="ex_nums.dfdl.xsd"
+    name="ex_nums_runtime2"
     config="config-runtime2">
     <tdml:document>
-      <tdml:documentPart type="file">ex_nums_parse.dat</tdml:documentPart>
+      <tdml:documentPart type="file">ex_nums.dat</tdml:documentPart>
     </tdml:document>
     <tdml:infoset>
-      <tdml:dfdlInfoset type="file">ex_nums_unparse_runtime2.xml</tdml:dfdlInfoset>
+      <tdml:dfdlInfoset type="file">ex_nums.dat.xml.runtime2</tdml:dfdlInfoset>
     </tdml:infoset>
   </tdml:parserTestCase>
 
   <tdml:unparserTestCase
-    name="ex_nums_unparse_runtime2"
-    model="ex_nums.dfdl.xsd"
-    config="config-runtime2">
-    <tdml:infoset>
-      <tdml:dfdlInfoset type="file">ex_nums_unparse_runtime2.xml</tdml:dfdlInfoset>
-    </tdml:infoset>
-    <tdml:document>
-      <tdml:documentPart type="file">ex_nums_parse.dat</tdml:documentPart>
-    </tdml:document>
-  </tdml:unparserTestCase>
-
-  <tdml:unparserTestCase
-    name="ex_nums_unparse_errors"
     model="ex_nums.dfdl.xsd"
+    name="ex_nums_runtime2_error"
     config="config-runtime2">
     <tdml:infoset>
-      <tdml:dfdlInfoset type="file">ex_nums_unparse_errors.xml</tdml:dfdlInfoset>
+      <tdml:dfdlInfoset type="file">ex_nums.dat.xml.runtime2.error</tdml:dfdlInfoset>
     </tdml:infoset>
     <tdml:errors>
       <tdml:error>value</tdml:error>
+      <tdml:error>boolean_false</tdml:error>
```

Comment:
I guess we missed this in the past, but it seems odd to get both an infoset and errors. I thought we even had a check that only allowed one or the other, but I guess not.

Though, you can get both validation errors and with an infoset, which is what I think thse are? We have a `<tdml:validatorErrors>` tag for that. I assume runtime2 just doesn't differentiate between the two types of errors?

Reply:
This is an `unparseTestCase`, not a `parseTestCase`, so that's why it has both an infoset and errors.  However, I've managed to make Runtime2DataProcessor return validation errors and change this test to a parse test case with a document, an infoset, and validation errors.  (I found out the hard way that an `unparseTestCase` isn't allowed to contain validation errors, unlike a `parseTestCase`.)  My next commit will have these changes.

Reply:
Ah yes, you're right. I didn't realize this was an unparse test. 

I wonder if unparseTestCase not supporting validation errors is a bug? runtime1 doesn't implement validation for unparse so it wouldn't surprise me if we never came across that.

Reply:
Yes, the TDML standard and TDMLRunner not supporting validation errors for unparse feels like a bug to me.  Even after I changed the test to a parseTestCase, I still couldn't allow the test case to do a round trip (parse, then unparse back to the same data file) because there was no way for Runtime2DataProcessor to return validation errors inside its UnparseResult.  Both parse and unparse runtime2 code report the same validation errors so the test case would have round-tripped fine with a document, an infoset, and the same validation errors for both parse and unparse.  However,  I don't know the process for making changes to the TDML standard (I believe the changes have to be accepted by the IBM folks and others, correct?).

Reply:
The unparser doesn't provide any validation, other than the validation inherent in unparsing. There's never a call to Xerces to validate, for example.